packages/astro/src/@types/astro.ts

```diff
@@ -1686,6 +1686,47 @@ export interface AstroUserConfig {
 	 */
 	legacy?: object;
 
+	/**
+	 * @docs
+	 * @name experimental.security
+	 * @type {boolean}
+	 * @default `{}`
+	 * @version 4.6.0
+	 * @description
+	 *
+	 * Enables different security measures inside an Astro website.
+	 *
+	 * ```js
+	 * // astro.config.mjs
+	 * export default defineConfig({
+	 *   output: "server",
+	 *   security: {
+	 *     checkOrigin: true
+	 *   }
+	 * })
+	 * ```
+	 */
+	security?: {
+		/**
+		 * @name security.checkOrigin
+		 * @type {boolean}
+		 * @default 'false'
+		 * @version 4.6.0
+		 * @description
+		 *
+		 * The CSRF protection works only for pages rendered on demand (SSR) using `server` or `hybrid` mode. The pages must opt out of prerendering in `hybrid` mode.
```

Comment:
I'm wondering about something like this...

``````suggestion
	 * Enables security measures, such as Cross-Site Request Forgery (CSRF) protection, for an Astro website.
	 *
	 * These features only exist for pages rendered on demand (SSR) using `server` or `hybrid` mode. The pages must opt out of prerendering in `hybrid` mode.
	 *
	 * ```js
	 * // astro.config.mjs
	 * export default defineConfig({
	 *   output: "server",
	 *   security: {
	 *     checkOrigin: true
	 *   }
	 * })
	 * ```
	 */
	security?: {
		/**
		 * @name security.checkOrigin
		 * @type {boolean}
		 * @default 'false'
		 * @version 4.6.0
		 * @description
``````
If it's true that all of the security measures that will eventually be here will require server mode, then I think it makes sense to front-load the SSR requirement at the top, then it won't need to be in each item below.

Plus, since the value is no longer named `csrfProtection`, we can't assume people will make the connection to CSFR protection and will have to spell it out a bit more somewhere. It's probably OK to have the description below be directly about checking origin, since that's the new name of the feature.

Reply:
I think your suggestion should be belong to `security.checkOrigin`. In the future, the `security` object might contain other security measures unrelated to CSRF.

Reply:
OK, what about the part about requiring SSR? Do you already know/think these will all require server-rendering? That was the thing I was mostly thinking about putting up front if everything was going to require it, so we wouldn't have to duplicate it for more items.

If not, it's totally fine to leave it where it was!

Reply:
> OK, what about the part about requiring SSR? Do you already know/think these will all require server-rendering?

I think, yeah, I doubt that a static build could have some security flaws; in the end, nothing is exposed other than some static file.